ern-api-impl-gen/src/ApiImpl.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+
+import fs from 'fs'
+import inquirer from 'inquirer'
+
+import {
+  Dependency,
+  shell
+} from 'ern-util'
+import {
+  ModuleTypes,
+  yarn
+} from 'ern-core'
+
+import ApiImplGen from './generators/ApiImplGen'
+
+const API_NAME_RE = /([^/]*)$/
+
+const path = require('path')
+
+export default async function generateApiImpl ({
+  apiDependency,
+  outputDirectory,
+  nativeOnly,
+  forceGenerate,
+  reactNativeVersion,
+  hasConfig = false,
+  paths
+} : {
+  apiDependency: Dependency,
+  outputDirectory: string,
+  nativeOnly: boolean,
+  forceGenerate: boolean,
```

Comment:
Looking in details at the implementation I can tell that `forceGenerate` is not required (by default it won't force generate).

However if I just look at this function object parameter I have no way to easily tell apart required vs optional properties (not from the object destructuring block itself nor from the flow typedef).  `flow` will show an error if `forceGenerate` is not provided by the caller.

You should typedef it as `forceGenerate?: boolean` instead, which makes it an optional object property. Same I think for `hasConfig`, you shouldn't use a `false` default, just type it as `hasConfig?: boolean` as well.  This way just by looking at the signature of the function (or looking at flow info when invoking the function) it's easy to distinguish required vs optional object properties.